app/services/search_service/exact_search.rb

```diff
@@ -58,7 +59,13 @@ def find_commodity(query)
 
       # NOTE: at the moment scope .declarable is not enough to
       # determine if it is really declarable or not
-      commodity.present? && commodity.declarable? ? commodity : nil
+      if commodity.blank?
+        nil
+      elsif commodity.declarable?
+        commodity
+      else
+        Subheading.actual.by_code(query).declarable.non_hidden.first
```

Comment:
Do we want take here?

```suggestion
        Subheading.actual.by_code(query).declarable.non_hidden.take
```

Reply:
Not sure how search plays with 404s

Reply:
And clearly nil is being used as not found

Reply:
So probably this is actually more consistent with this particular api

Reply:
No, we don't want to raise at this point - this needs to return nil - the caller uses `find_commodity || find_heading` and has its own handling for if there's no result